common/lib/xmodule/xmodule/tabs.py

```diff
@@ -168,610 +161,30 @@ def from_json(tab_dict):
-        super(TextbookTabsBase, self).__init__(
-            name=_("Textbooks"),
-            tab_id=tab_id,
-            link_func=None,
-        )
-
-    @abstractmethod
-    def items(self, course):
-        """
-        A generator for iterating through all the SingleTextbookTab book objects associated with this
-        collection of textbooks.
-        """
-        pass
-
-
-class TextbookTabs(TextbookTabsBase):
-    """
-    A tab representing the collection of all textbook tabs.
-    """
-    type = 'textbooks'
-
-    def __init__(self, tab_dict=None):  # pylint: disable=unused-argument
-        super(TextbookTabs, self).__init__(
-            tab_id=self.type,
-        )
-
-    def can_display(self, course, settings, is_user_authenticated, is_user_staff, is_user_enrolled):
-        return settings.FEATURES.get('ENABLE_TEXTBOOK')
-
-    def items(self, course):
-        for index, textbook in enumerate(course.textbooks):
-            yield SingleTextbookTab(
-                name=textbook.title,
-                tab_id='textbook/{0}'.format(index),
-                link_func=lambda course, reverse_func, index=index: reverse_func(
-                    'book', args=[course.id.to_deprecated_string(), index]
-                ),
-            )
-
-
-class PDFTextbookTabs(TextbookTabsBase):
-    """
-    A tab representing the collection of all PDF textbook tabs.
-    """
-    type = 'pdf_textbooks'
-
-    def __init__(self, tab_dict=None):  # pylint: disable=unused-argument
-        super(PDFTextbookTabs, self).__init__(
-            tab_id=self.type,
-        )
-
-    def items(self, course):
-        for index, textbook in enumerate(course.pdf_textbooks):
-            yield SingleTextbookTab(
-                name=textbook['tab_title'],
-                tab_id='pdftextbook/{0}'.format(index),
-                link_func=lambda course, reverse_func, index=index: reverse_func(
-                    'pdf_book', args=[course.id.to_deprecated_string(), index]
-                ),
-            )
-
-
-class HtmlTextbookTabs(TextbookTabsBase):
-    """
-    A tab representing the collection of all Html textbook tabs.
-    """
-    type = 'html_textbooks'
-
-    def __init__(self, tab_dict=None):  # pylint: disable=unused-argument
-        super(HtmlTextbookTabs, self).__init__(
-            tab_id=self.type,
-        )
-
-    def items(self, course):
-        for index, textbook in enumerate(course.html_textbooks):
-            yield SingleTextbookTab(
-                name=textbook['tab_title'],
-                tab_id='htmltextbook/{0}'.format(index),
-                link_func=lambda course, reverse_func, index=index: reverse_func(
-                    'html_book', args=[course.id.to_deprecated_string(), index]
-                ),
+        # TODO: don't import openedx capabilities from common
+        from openedx.core.djangoapps.course_views.course_views import CourseViewTypeManager
```

Comment:
There are two levels of modularity problems with this PR:
1. There are a couple of places in common/lib/xmodule that need to refer to Open edX Django apps.
2. There are a few view types that live in LMS but which are need in Studio. This works today as Studio picks up all the registered entry points, but won't work so well if we ever split Studio and LMS.

We don't currently have a good story for how either of these should really work. We did this whole refactoring so that the course view types would be able to use ``has_access`` et al, but those features are only in the LMS. We could split course view types into two entry points (the metadata that is needed by both Studio and LMS, and then the runtime that is used in the LMS), but that seems confusing for the developer.

Regarding the imports of openedx from common, I haven't come up with a good option. I looked into moving all of this code into common/djangoapps, but that seemed too disruptive. There are already 49 such imports in common, so it seemed reasonable to just go with the flow here.

What would you recommend?